connectivity/builder/local_redirect_policy_with_nodedns.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package builder
+
+import (
+	_ "embed"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/connectivity/tests"
+	"github.com/cilium/cilium-cli/utils/features"
+)
+
+//go:embed manifests/client-egress-node-local-dns.yaml
+var clientEgressNodeLocalDNSYAML string
+
+type localRedirectPolicyWithNodeDNS struct{}
+
+func (t localRedirectPolicyWithNodeDNS) build(ct *check.ConnectivityTest, templates map[string]string) {
+	newTest("local-redirect-policy-with-node-dns", ct).
+		WithCondition(func() bool { return ct.Params().IncludeUnsafeTests }).
+		WithCiliumPolicy(templates["clientEgressNodeLocalDNSYAML"]).
+		WithFeatureRequirements(
+			features.RequireEnabled(features.NodeLocalDNS),
+			features.RequireEnabled(features.NodeWithoutCilium),
```

Comment:
Why does it need a node without cilium?

Reply:
That's because local-redirect-policy-with-node-dns checks if the clients can access `echo-external-node.cilium-test.svc.cluster.local` on the node without Cilium to make sure that the node DNS with LRP is compatible with the DNS proxy.

```
    - toFQDNs:
      - matchName: "echo-external-node.{{.TestNamespace}}.svc.cluster.local"
```